proposals/lip-0048.md

```diff
@@ -25,32 +25,43 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
 
 ## Rationale
 
-The fee structure presented here serves to allow defining sufficient cost to manage state growth and prevent transaction pool spamming. In principle, any transaction resulting in a state change should incur a fee. This reasoning is implemented in the `payFee` method, which can be invoked by other modules.
+### Initialization Fee
```

Comment:
In my opinion, starting with this topic is a bit confusing. Moreover, not all extra-command fees are really initialization fees, for example the sidechain registration fee. 

I would suggest to start with the transaction fee subsection. 

We could even link to the transaction schema LIP and start by saying that each transaction has a fee.

Reply:
Suggested structure (unpolished):

Subsection: Transaction Fees.

In the lisk ecosystem, as in almost all blockchains, each transaction has an associated transaction fee (link transaction schema LIP 68). Transaction fees prevent spamming a blockchain with unlimited transactions at no cost. Moreover, part of the transaction fees is provided as reward to validators who run and maintain the security of the network, as we explain in detail below.  [this part makes clear *why* we have transaction fees].

[Then add the sub-subsections of FeeTokenID and Minimum Fee Per Transaction, which are very well written IMO. ]

Subsection: Initialization fees ( we could even call them state growth fees or something else)

The fee structure presented here allows for defining a sufficient cost to manage state growth. In principle, any transaction resulting in a state increase should imply a fee.  This reasoning is implemented in the [`payFee`](#payfee) method, which can be invoked by other modules to charge an additional fee if during the execution of a transaction, a state growth is trigerred (e.g., new validator register in PoS module, new sidechain gets registered, or initialize a new user account).

[then discuss what happens after trs execution]

We call the part of the fee used during the execution of a transaction (i.e., the sum of minFee plus the initialization fees) as the "consumed" part of the fee;  the rest of the fee is called available fee and it is given as reward to the block generator, as an additional reward for securing the network. This way, it is expected that whenever the network is congested, validators prioritize transactions with higher available fee. 

[Add here the Fee pool paragraph]

[Add the ccm paragraph]

Reply:
Yes, I also noticed that starting with "initialization" fees (which are, I agree, not only that, so the naming is not perfect) is a bit awkward, but I had no idea where to introduce them so I thought the beginning of the Rationale section is OK. 
Thanks for the suggestions, it's better this way since transaction fees are discussed first, which is a more natural thing to do.